db/revision_cache_bypass.go

```diff
@@ -0,0 +1,55 @@
+package db
+
+import "github.com/pkg/errors"
+
+// BypassRevisionCache is an implementation of the RevisionCache interface that does perform any caching.
+// For any Get operation, it will always immediately fetch the requested revision from the bucket.
+type BypassRevisionCache struct {
+	loaderFunc RevisionCacheLoaderFunc
+}
+
+// Get fetches the revision for the given docID and revID immediately from the bucket.
+func (rc *BypassRevisionCache) Get(docID, revID string, copyType BodyCopyType) (docRev DocumentRevision, err error) {
+	if rc.loaderFunc == nil {
+		return DocumentRevision{}, errors.New("nil loaderFunc")
+	}
+
+	docRev.RevID = revID
+	docRev.Body, docRev.History, docRev.Channels, docRev.Attachments, docRev.Expiry, err = rc.loaderFunc(IDAndRev{DocID: docID, RevID: revID})
+	if err != nil {
+		return DocumentRevision{}, err
+	}
+
+	return docRev, nil
+}
+
+// GetActive fetches the active revision for the given docID immediately from the bucket.
+func (rc *BypassRevisionCache) GetActive(docID string, context *DatabaseContext, copyType BodyCopyType) (docRev DocumentRevision, err error) {
+	doc, err := context.GetDocument(docID, DocUnmarshalAll)
+	if err != nil {
+		return DocumentRevision{}, err
+	}
+
+	docRev.RevID = doc.CurrentRev
+	docRev.Body, docRev.History, docRev.Channels, docRev.Attachments, docRev.Expiry, err = context.revCacheLoaderForDocument(doc, doc.syncData.CurrentRev)
```

Comment:
Between the two of them, revCacheLoader and revCacheLoader for document only use the following methods from databaseContext:
- context.GetDocument
- context.getRevision
If we want to decouple the rev cache from DatabaseContext, defining an interface of those two methods might be the way to go.  We could then move the revCacheLoader and revCacheLoaderForDocument code somewhere that could be shared by all rev cache implementations, and customized for test.

(as discussed, that could be a followup PR)